src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -24,13 +24,16 @@ public LoggerAdapter(IEnumerable<ILogger> loggers, ITelemetryReporter? telemetry
         _telemetryReporter = telemetryReporter;
     }
 
-    public IDisposable BeginScope<TState>(TState state)
+    public IDisposable? BeginScope<TState>(TState state) where TState : notnull
     {
         var compositeDisposable = new CompositeDisposable();
         foreach (var logger in _loggers)
         {
             var disposable = logger.BeginScope(state);
-            compositeDisposable.AddDisposable(disposable);
+            if (disposable != null)
+            {
+                compositeDisposable.AddDisposable(disposable!);
```

Comment:
I don't think the `!` should be needed after the null check

Reply:
```suggestion
                compositeDisposable.AddDisposable(disposable);
```